src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1381,6 +1390,124 @@ void java_bytecode_parsert::rclass_attribute(classt &parsed_class)
   {
     rRuntimeAnnotation_attribute(parsed_class.annotations);
   }
+  else if(attribute_name == "BootstrapMethods")
+  {
+    INVARIANT(
+      !parsed_class.read_attribute_bootstrapmethods,
+      "only one BootstrapMethods argument is allowed in a class file");
+
+    // mark as read in parsed class
+    parsed_class.read_attribute_bootstrapmethods = true;
+    u2 num_bootstrap_methods = read_u2();
+    for(size_t i = 0; i < num_bootstrap_methods; i++)
+    {
+      u2 bootstrap_methodhandle_ref = read_u2();
+      const pool_entryt &entry = pool_entry(bootstrap_methodhandle_ref);
+      u2 num_bootstrap_arguments = read_u2();
```

Comment:
Rather than reading the arguments below - I'd be tempted to read them all now into an appropriate vector so there's no chance of missing a byte (e.g. if we add more `handle_type`s)